Add OAuth documentation and bump version to 1.2.0 (#349)
Volue-Public/energy-mesh-python: docs/source/authentication.rst

=====================
Client authentication
=====================

> Mesh server could be configured to protect its gRPC API so that only
> authenticated users that are assigned to specific roles can access it. There
> are two options:

Reviewer: ```suggestion
A Mesh server may be configured to protect its gRPC API so that only
authenticated users that are assigned to specific roles can access it. There
are two options:
```

>   - **Kerberos** where Mesh authenticates users and grants access to specific
>     APIs based on what AD groups the authenticated users belong to.

Reviewer: ```suggestion
  - **Kerberos,** where Mesh authenticates users and grants access to specific
    APIs based on what AD groups the authenticated users belong to.
```

>   - **External access token** where users authenticate and request access to
>     Mesh in external authorization servers, and Mesh validates the access
>     token that is obtained from the authorization server.

Reviewer: ```suggestion
  - **External access token,** where users authenticate and request access to
    Mesh from external authorization servers, and Mesh validates the access
    token that is obtained from the authorization server.
```
